modules/commons/src/main/java/org/apache/synapse/commons/json/JsonUtil.java

```diff
@@ -521,16 +528,23 @@ public static OMElement getNewJsonPayload(MessageContext messageContext, InputSt
         }
         boolean isObject = false;
         boolean isArray = false;
+        boolean isValue = false;
         if (inputStream != null) {
             InputStream json = toReadOnlyStream(inputStream);
             messageContext.setProperty(ORG_APACHE_SYNAPSE_COMMONS_JSON_JSON_INPUT_STREAM, json);
             // read ahead few characters to see if the stream is valid...
             boolean isEmptyPayload = true;
             boolean valid = false;
             try {
-                // check for empty/all-whitespace streams
+                /*
+                * Checks for all empty or all whitespace streams and if found  sets isEmptyPayload to false. The while
+                * loop exits if one of the valid literals is found. If no valid literal is found the it will loop to
+                * the end of the stream and the next if statement after the loop makes sure that valid will remain false
+                * */
                 int c = json.read();
-                while (c != -1 && c != '{' && c != '[') {
+                while (c != -1 && c != '{' && c != '[' && c != '"' && c != 't' && c != 'n' && c != 'f' && c != '-' &&
```

Comment:
Can we add all the valid lieterals into a set and check if the charcter is included in it? Would it be a better approach?

Reply:
Fixed in #759